astropy/utils/diff.py

```diff
@@ -0,0 +1,155 @@
+import difflib
+import functools
+import operator
+from functools import reduce
+from itertools import islice
+
+import numpy as np
+
+from .misc import indent
+
+__all__ = ['fixed_width_indent', 'diff_values', 'report_diff_values',
+           'where_not_allclose']
+
+# Smaller default shift-width for indent
+fixed_width_indent = functools.partial(indent, width=2)
+
+
+def diff_values(a, b, rtol=0.0, atol=0.0):
+    """
+    Diff two scalar values. If both values are floats, they are compared to
+    within the given absolute and relative tolerance.
+
+    Parameters
+    ----------
+    a, b : int, float, str
+        Scalar values to compare.
+
+    rtol, atol : float
+        Relative and absolute tolerances as accepted by
+        :func:`numpy.allclose`.
+
+    Returns
+    -------
+    is_different : bool
+        `True` if they are different, else `False`.
+
+    """
+    if isinstance(a, float) and isinstance(b, float):
+        if np.isnan(a) and np.isnan(b):
+            return False
+        return not np.allclose(a, b, rtol=rtol, atol=atol)
+    else:
+        return a != b
+
+
+def report_diff_values(fileobj, a, b, ind=0):
```

Comment:
For ease of use, `sys.stdout` could be used as default value for `fileobj` ?

Reply:
This requires `fileobj` to be moved to after `a, b` as a keyword argument, no?

I guess since we're moving the function to a whole different module, function signature change at this point is probably an acceptable risk. Just want to make sure this is really what you want. (Same comment for the `ind -> indent_width` request below.)

Reply:
I actually like this, that is moving `fileobj` to 3rd position.

Reply:
No, I love it!

Reply:
Yes, it means changing the API. Since we are making this function public it's the good time to think about a better signature.

Reply:
Of course changing a innocent looking API call broke FITS diff somehow. Investigating...

Reply:
OK got it. *angry fist at recursive calls*

Reply:
ohh, any innocent change (it doesn't need to be API chance) can brake it, so I feel your pain 😱